### ISSUE TITLE
Very slow UI redraw and white fringes on hidpi Linux

### ISSUE THREAD

**Issue by nlaplante** [atom/atom#10518]
When running Atom on my 4k laptop monitor, the UI redraws are very slow and there are white fringes all over. 

I tried to disable each 3rd party package I had but I get the same behavior each time.

 * Resolution: 3840 x 2160
 * Gnome 3.18
 * Fedora 23
 * Linux 4.3.3-300.fc23.x86_64  GNU/Linux
 * Atom 1.4.0

```
$ egrep -i " connected|card detect|primary dev|Setting driver" /var/log/Xorg.0.log
[   146.111] (II) intel(0): Using Kernel Mode Setting driver: i915, version 1.6.0 20150731
```

```
$ glxinfo | grep -i vendor
server glx vendor string: SGI
client glx vendor string: Mesa Project and SGI
OpenGL vendor string: Intel Open Source Technology Center
```

**Comment by icelight0** [comment]
Hi,
i'm having the same issue since 1.2 i think. Rolling back to 1.1 fixes all problems, but isn't really a permanent solution. I think it's an issue with chromium embedded, as vivaldi had the same issues until very recently. It works at least until 2560x1440, but not at 3840x2160.

`% egrep -i " connected|card detect|primary dev|Setting driver" /var/log/Xorg.0.log                                                                      :(
[  2857.172] (II) intel(0): Using Kernel Mode Setting driver: i915, version 1.6.0 20150731`

`% glxinfo | grep -i vendor
server glx vendor string: SGI
client glx vendor string: Mesa Project and SGI
    Vendor: Intel Open Source Technology Center (0x8086)
OpenGL vendor string: Intel Open Source Technology Center

**Comment by nlaplante** [comment]
Atom 1.4.1 exhibits the same behavior

**Comment by nlaplante** [comment]
See behavior:
[![Video showing behavior](http://img.youtube.com/vi/ZzZYhCxxu9E/0.jpg)](http://www.youtube.com/watch?v=ZzZYhCxxu9E)

**Comment by nlaplante** [comment]
It render slower and slower as the opened file grows

**Comment by alexmherrmann** [comment]
I'm having the same issue. For me it is literally as simple as moving my Atom window from the 4k display to the FHD screen. Unfortunately I have the same setup as nlaplante with a 4k screen on the laptop (I only ever run it in 2880 x 1620 however), so I have to either down-rez it to 1080p so that Atom runs usably, or just don't use Atom. Would love to see the speed back to the pre-update speed. I update atom infrequently, but I want to say the problem started around 1.1-1.2.

**Comment by alexmherrmann** [comment]
I'm having the same issue. For me it is literally as simple as moving my Atom window from the 4k display to the FHD screen. I don't have quite the same issues with white boxes everywhere, but I also only run it at 2880x1620 max, but it does do the weird scanline thing too.

**Comment by nasedil** [comment]
I confirm, same stuff after update on Ubuntu 15.10 with xfce.  It was a bit better before I've updated Atom today to the latest version.

Also some parts remain white.  And sometimes different areas of screen are erased while other parts of screen are redrawn when I move mouse around.

**Comment by leifcr** [comment]
Other related issues: #9724, #10644 and #9561

**Comment by nlaplante** [comment]
Just to add that this could be a problem with Electron, since Git Kraken suffers from the same issue.

**Comment by nlaplante** [comment]
Starte a Wayland session (Xwayland) and Atom seems alot faster, though not as fast a when running in lower resolutions (1080p).

**Comment by willotter** [comment]
I am experiencing the same issue. I had it on both Debian Stretch and on Arch. Using Gnome 3.18. Atom 1.5.4

Using Wayland does seem to give an improvement.

**Comment by willotter** [comment]
@nlaplante I found passing `--ignore-gpu-blacklist --disable-gpu-sandbox` options when calling atom fixed the majority of my issues. They don't appear to be documented but I think they are passed by electron to the chromium process.

**Comment by nlaplante** [comment]
@willotter I tried this and I had the same slow behavior on 1.5.4

**Comment by saward** [comment]
@nlaplante A quick test indicates to me that this improves performance significantly.  Do you know what the significance of these options are?  What it is they are doing?

**Comment by willotter** [comment]
@saward I am not 100% to be honest. My understanding is that sometimes you can have GPUs blacklisted by the kernel (I am running bumblebee which does this I believe). These flags allow chromium to bypass this and thus enable hardware acceleration. Someone else with more understanding of the inner workings of chromium might be able to give a better explanation.

**Comment by saward** [comment]
@willotter Interesting.  When I run atom with these options, and try 'optirun --status', it reports that my discrete video card is off.  I don't know whether optirun can detect discrete card being on if it's started by something else though.

Maybe these options allow it to use the intel gpu?

**Comment by willotter** [comment]
@saward That would make sense. Interestingly I don't have this problem in Chrome .

**Comment by 50Wliu** [comment]
@willotter @saward @nlaplante Is there any chance you can try building master and seeing if the issues are resolved?  Master is running Chrome 47, whereas 1.5 is running (I believe) Chrome 45.

**Comment by willotter** [comment]
@50Wliu I will do that and let you know how I get on.

**Comment by saward** [comment]
@50Wliu still slow using 1.6.0-beta7 (linux), and fast enough when using the options that @willotter provided.

In short: no difference.

**Comment by 50Wliu** [comment]
@saward 1.6.0-beta7 is also running Chrome 45 :wink:.  We only managed to upgrade to 47 a few weeks ago.

**Comment by karalabe** [comment]
Just updated to 1.6, from an ancient version. Sad to see that I need to pass extra params to work around the issue, especially since it's been present for quite a long time now :P

**Comment by 50Wliu** [comment]
If anyone can try 1.7.0-beta0, that would be great.

**Comment by karalabe** [comment]
@50Wliu I actually tried both stable and the beta. Both behaved exactly the same way.

**Comment by betterphp** [comment]
@willotter Thanks that fixed it for me using 1.6.2 on Ubuntu 15.10

I changes the Exec line of `/usr/share/applications/atom.desktop` to be
~~~
Exec=/opt/atom/atom --ignore-gpu-blacklist --disable-gpu-sandbox %F
~~~
to make the launcher icon work nicely

**Comment by willotter** [comment]
@50Wliu I am now running 1.7.2 and it seems to be working without the additional cli parameters. So well done :+1:

**Comment by karalabe** [comment]
Just tried 1.7.2, and the slow redraws are still there for me. It takes about 2-3 seconds for the entire screen to be filled, kind of like downloading an image over dialup :P

**Comment by cbuschardt** [comment]
I'm on 1.7.2 and if the windows is wider than about 3200 pixels suddenly I get flickering, slow drawing, and display corruption with some plugins.   It appears to be an electron/chromium issue.
I've tried with both Intel and NVidia.

**Comment by cbuschardt** [comment]
This option fixes all rendering issues for me.  

atom --ignore-gpu-blacklist

**Comment by xfenix** [comment]
3840x2160 msi gs60 ghost pro 4k - same problem for me. On Ubuntu 16.04. Atom 1.7.4 fresh install.
Fixed with betterphp solution perfectly. Thanks a lot.

**Comment by 50Wliu** [comment]
I have another request...if anyone wants to try building the latest master and seeing if they can reproduce this issue, that would be appreciated.  We recently upgraded the Chrome version again which might fix a few more of these reports.

There's really nothing we can do about this though other than upgrading Chrome and hoping for the best, since Chrome decides which GPUs are blacklisted.

**Comment by shearn89** [comment]
@50Wliu - happy to try and build this evening when I get home from work.

**Comment by shearn89** [comment]
@50Wliu - had a spare minute! Just built from HEAD git (0e0150a0438396ccb0a36a849a804b91d54b369a) and the issue appears to work. Comparing with the latest .deb build from the website: changing the theme would show the slow redraw issue, whereas from the source build it does not.

**Comment by morhook** [comment]
On my Dell XPS 15 9550 4k the issue was happening (and could be fixed with mentioned parameters `--ignore-gpu-blacklist --disable-gpu-sandbox`). Just downloaded https://github.com/atom/atom/releases/tag/v1.9.0-beta0 and with this one the problem is not happening!!

Thanks a lot @50Wliu and @shearn89 !

**Comment by 50Wliu** [comment]
Awesome, thanks for the updates!  @nlaplante Can you still reproduce your problem on Atom 1.9.0?

**Comment by karalabe** [comment]
Seems to work fine for me too (beta) :D

**Comment by nlaplante** [comment]
I'll have a look when I get some time hopefully in the next days. I'm
covered with work for now :/

*--*
*Sent from Gmail web*

**Comment by mikebabb** [comment]
@willotter you are a hero, thank you so much for your arguments fix! This was driving me crazy.

**Comment by morhook** [comment]
I think this should be closed, I've installed atom released 1.9.6 and it's working fine! Thanks everyone involved also.

**Comment by 50Wliu** [comment]
Thanks for the responses everyone!  Glad to hear that it's fixed now! :tada:
